### PR TITLE
Added ability to choose workspace

### PR DESCRIPTION
**Pull request by ghost**
*need to implement getting info from db and sending token to backend*

### CODE REVIEW

File: frontend/src/components/workspaces/workspaces.tsx [hunk 1]
```diff
@@ -1,3 +1,91 @@
...
+  const [popUpText, setPopUpText] = useState('');
+
+  const [cookies, setCookie, removeCookie] = useCookies(['workspaceID']);
+
+  const history = useHistory();
+
+  useEffect(() => {
+    dispatch(workspacesActions.loadWorkspaces());
+    const workspaceID = cookies.workspaceID;
+    if (workspaceID) dispatch(workspacesActions.SetCurrentWorkspaceID(workspaceID));
```
Comment (aleksandr-kuharenko) [inline frontend/src/components/workspaces/workspaces.tsx:26]: [B12](https://github.com/BinaryStudioAcademy/quality-criteria/blob/production/source/javascript.md). Curly braces are required everywhere.

File: frontend/src/components/workspaces/workspaces.tsx [hunk 2]
```diff
@@ -1,3 +1,91 @@
...
+  }, []);
+
+  useEffect(() => {
+    if (currentWorkspaceID) {
+      setCookie('workspaceID', currentWorkspaceID);
```
Comment (aleksandr-kuharenko) [inline frontend/src/components/workspaces/workspaces.tsx:31]: move `workspaceId` to Enum

File: frontend/src/components/workspaces/components/workspace-item/workspace-item.tsx [hunk 3]
```diff
@@ -1,3 +1,14 @@
-const Workspaces: React.FC = () => <h1>Workspace item</h1>;
+import { IWorkspaceCreation } from 'common/interfaces/workspace';
+import Card from 'react-bootstrap/Card';
+import './styles.scss';
 
-export default Workspaces;
+const WorkspaceItem: React.FC<IWorkspaceCreation> = ({ title }) =>
+  (
+    <Card className="workspace-card shadow-sm rounded bg-white border-0">
```
Comment (aleksandr-kuharenko) [inline frontend/src/components/workspaces/components/workspace-item/workspace-item.tsx:7]: Click on workspace should update workspaceId in the cookie and open /pages

File: backend/src/services/workspace.service.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  const userWorkspaceRepository = new UserWorkspaceRepository();
+  const refreshTokenRepository = new RefreshTokenRepository();
+  const { userId } = await refreshTokenRepository.findOne({ token });
+  const { workspacesId } = await userWorkspaceRepository.find({ userId });
+  const workspaces = [] as IWorkspace[];
+  for (const workspaceId of workspacesId) {
+    const workspace = await workspaceRepository.findOne({ workspaceId });
+    workspaces.push({ ...workspace, title: workspace.name });
+  }
+  return workspaces;
```
Comment (aleksandr-kuharenko) [inline backend/src/services/workspace.service.ts:19]: From frontend, you receive `accessToken` which is not stored in DB, but contains userId in payload. There will be a middleware that will add userId to `req`. (should be implemented in context of this [ticket](https://trello.com/c/cWEtXNCm/34-implement-middlewares-to-secure-server))

You need to find only User Entity

**Review by aleksandr-kuharenko — CHANGES_REQUESTED** [review]

File: backend/package.json [hunk 5]
```diff
@@ -26,13 +26,15 @@
   "devDependencies": {
     "@types/cors": "2.8.12",
     "@types/cryptr": "4.0.1",
-    "nodemon": "2.0.6",
-    "ts-node-dev": "1.0.0",
-    "tscpaths": "0.0.9",
+    "@types/jsonwebtoken": "^8.5.4",
+    "@types/uuid": "^8.3.1",
```
Comment (aleksandr-kuharenko) [inline backend/package.json:30]:
```suggestion
    "@types/jsonwebtoken": "8.5.4",
    "@types/uuid": "8.3.1",
```

File: frontend/src/services/workspace-api/workspace-api.service.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  private _http: Http;
+
+  constructor() {
+    this._http = new Http;
+  }
+
+  // TODO send token
+
+  public async create(payload: IWorkspaceCreation): Promise<IWorkspace> {
+    return this._http.load('http://localhost:3000/api/workspaces', {
```
Comment (aleksandr-kuharenko) [inline frontend/src/services/workspace-api/workspace-api.service.ts:15]:
```suggestion
    return this._http.load('/workspaces', {
```

File: frontend/src/services/workspace-api/workspace-api.service.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+      method: HttpMethod.POST,
+      payload: JSON.stringify(payload),
+      contentType: ContentType.JSON,
+    });
+  }
+  public async get(): Promise<IWorkspace[]> {
+    return this._http.load('http://localhost:3000/api/workspaces');
```
Comment (aleksandr-kuharenko) [inline frontend/src/services/workspace-api/workspace-api.service.ts:22]:
```suggestion
    return this._http.load('/workspaces');
```

**Review by aleksandr-kuharenko — COMMENTED** [review]
Add private method to http.service, which will add access token from local storage to headers

**Review by aleksandr-kuharenko — CHANGES_REQUESTED** [review]
PR with middlewares merged into `develop`.
Please update your branch

**Review by sofiia-chorna — DISMISSED** [review]

File: backend/package.json [hunk 8]
```diff
@@ -55,6 +56,7 @@
     "reflect-metadata": "0.1.13",
     "ts-node": "9.0.0",
     "typeorm": "0.2.34",
-    "typescript": "4.1.2"
+    "typescript": "4.1.2",
+    "uuid": "8.3.2"
```
Comment (aleksandr-kuharenko) [inline backend/package.json:60]: Not needed

**Review by aleksandr-kuharenko — CHANGES_REQUESTED** [review]

**Review by aleksandr-kuharenko — APPROVED** [review]

**Pull request merged**